### PR TITLE
7. Post comment by article ID - Tested

### PR DESCRIPTION
**Pull request by cgtorrubia**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -129,6 +129,47 @@ describe('GET /api/articles', () => {
...
+	test('POST - Status: 404 - responds with an error if user not found', () => {
+		return request(app)
+			.post('/api/articles/1/comments')
+			.send({
+				username: 'icellusedkars',
+			})
+			.expect(400)
+			.then(({ body }) => {
+				expect(body.msg).toBe('Bad request: A comment is required')
+			})
```
Comment (johnhutcheon) [inline __tests__/app.test.js:270]: We'd expect a 400 in this instance, as it's missing required fields (comment body in this case)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -129,6 +129,47 @@ describe('GET /api/articles', () => {
...
+				username: 'Demiurge',
+				body: 'This is a test comment, I am existing briefly to prove the existence of this endpoint.',
+			})
+			.expect(404)
+			.then(({ body }) => {
+				expect(body.msg).toBe('User not found')
+			})
+	})
+})
+
```
Comment (johnhutcheon) [inline __tests__/app.test.js:312]: Just a few more tests we need here please. Could we add the following?
- Status 201, ignores unnecessary properties (i.e. if you put something else apart from username and body in the post)
- Status 400, invalid ID, e.g. string of "not-an-id"
- Status 404, non existent ID, e.g. 0 or 9999

File: models/articles.models.js [hunk 3]
```diff
@@ -30,3 +30,28 @@ exports.selectArticles = () => {
 		return result.rows
 	})
 }
+
+exports.addCommentByArticleID = (articleComment, articleID) => {
+	const { username, body } = articleComment
+	if (!body) {
+		return Promise.reject({ status: 400, msg: 'Bad request: A comment is required' })
+	}
```
Comment (johnhutcheon) [inline models/articles.models.js:62]: We should also be checking that a username has been sent here as well

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Nice work! Just see my comments, a few changes required :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const data = require('../db/data/test-data')
 const db = require('../db/connection')
 const app = require('../app')
 const endpoints = require('../endpoints.json')
+const { expect } = require('@jest/globals')
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: Looks like this is an auto-add, not sure that you need it.

File: app.js [hunk 5]
```diff
@@ -30,14 +33,24 @@ app.use((err, req, res, next) => {
 	}
 })
 
+app.use((err, req, res, next) => {
```
Comment (Geo-Beard) [inline app.js]: You can have multiple if statements in one app.use - usually a good idea to keep PSQL errors together, but no problem leaving them separate.

**Review by Geo-Beard — COMMENTED** [review]
Looks OK overall - couple of small things, make those changes then feel free to merge and move on!

**Pull request merged**